File: 811-subdomain-visit-count/811-subdomain-visit-count.py
class Solution:
    def subdomainVisits(self, cpdomains: List[str]) -> List[str]:
        
        domain_count = {}
        res = []
        
        for string in cpdomains:
            count = string.split(" ")[0]
            domain = string.split(" ")[1]
            
            # print(string.split(" "))
            # print(count)
            # print(domain)
            
            while len(cpdomains) > 0:
                if domain not in domain_count:
                    domain_count[domain] = int(count)
                else:
                    domain_count[domain] += int(count)
                    
                if "." not in domain:
                    break;
                
                domain = domain[domain.index(".") + 1: ]
                # print(domain)
        
        for ele in domain_count:
            res.append(str(domain_count[ele]) + " " + ele)
        return res

File: 811-subdomain-visit-count/811-subdomain-visit-count.js
/**
 * @param {string[]} cpdomains
 * @return {string[]}
 */
var subdomainVisits = function(cpdomains) {
    
    const domainCount = {};
    
    for (let i = 0; i < cpdomains.length; i++){
        
        let count = cpdomains[i].split(" ")[0]
        let domain = cpdomains[i].split(" ")[1]
        
        // console.log(count)
        // console.log(domain)
        
        while(1){
            if(!domainCount[domain]){
                domainCount[domain] = Number(count)
            } else {
                domainCount[domain] += Number(count)
            }
            
             if(!domain.includes(".")) {
                break
            }
            
            domain = domain.substr(domain.indexOf(".") + 1)
            console.log(domain)
        }
    }
    
    // console.log(domainCount)
    
    const res = []
    
    for (let ele in domainCount){
        res.push(domainCount[ele] + " " + ele)
    }
    return res
};